File: raft/src/main/java/raft/AppendEntry.java
package raft;

import java.util.List;

public class AppendEntry {
    long term;
    RaftServer leader;
    //possibly club together as logentry
    long previousLogIndex;
    long previousLogTerm;
    List<LogEntry> entries;
    long leaderCommitIndex;

    public AppendEntry(long term, RaftServer leader, long previousLogIndex, long previousLogTerm, List<LogEntry> entries, long leaderCommitIndex) {
        this.term = term;
        this.leader = leader;
        this.previousLogIndex = previousLogIndex;
        this.previousLogTerm = previousLogTerm;
        this.entries = entries;
        this.leaderCommitIndex = leaderCommitIndex;
    }

    public long getTerm() {
        return term;
    }

    public RaftServer getLeader() {
        return leader;
    }

    public long getPreviousLogIndex() {
        return previousLogIndex;
    }

    public long getPreviousLogTerm() {
        return previousLogTerm;
    }

    public List<LogEntry> getEntries() {
        return entries;
    }

    public long getLeaderCommitIndex() {
        return leaderCommitIndex;
    }
}

File: raft/src/main/scala/com/raft/AppendEntry.scala
package com.raft

class
package raft

import java.util
import java.util.List

import raft.{LogEntry, RaftServer}


class AppendEntry(var term: Long, var leader: RaftServer, //possibly club together as logentry
                  var previousLogIndex: Long, var previousLogTerm: Long, var entries: util.List[LogEntry], var leaderCommitIndex: Long) {
  def getTerm: Long = term

  def getLeader: RaftServer = leader

  def getPreviousLogIndex: Long = previousLogIndex

  def getPreviousLogTerm: Long = previousLogTerm

  def getEntries: util.List[LogEntry] = entries

  def getLeaderCommitIndex: Long = leaderCommitIndex
}

